File: ai-agent-system/mcp-server           # Python-based Master Control Program/Assets/Scripts/Communication # Communication with server/Protocol.cs
using System;
using System.Collections.Generic;
using Newtonsoft.Json;

namespace Communication
{
    public static class Protocol
    {
        public static string SerializeMessage(Message message)
        {
            return JsonConvert.SerializeObject(message);
        }

        public static Message DeserializeMessage(string json)
        {
            return JsonConvert.DeserializeObject<Message>(json);
        }
    }

    [Serializable]
    public class Message
    {
        public string MessageType { get; set; }
        public Dictionary<string, object> Data { get; set; }
    }
}

File: ai-agent-system/mcp-server           # Python-based Master Control Program/src/communication # Communication protocols/protocol.py
def serialize_message(message):
    import json
    return json.dumps(message)

def deserialize_message(message_str):
    import json
    return json.loads(message_str)

def create_message(message_type, payload):
    return {
        "type": message_type,
        "payload": payload
    }

def parse_message(message):
    message_type = message.get("type")
    payload = message.get("payload")
    return message_type, payload